chainer/datasets/image_dataset.py

```diff
@@ -59,6 +59,8 @@ def get_example(self, i):
         path = os.path.join(self._root, self._paths[i])
         with Image.open(path) as f:
             image = numpy.asarray(f, dtype=self._dtype)
+        if len(image.shape) == 2:
```

Comment:
I believe `if image.ndim == 2` is the more common, please use it for readability. Could you also comment `# image is greyscale` on the next line?